src/qt/bitcoin.cpp

```diff
@@ -467,8 +467,7 @@ void BitcoinApplication::initializeResult(bool success)
     qDebug() << __func__ << ": Initialization result: " << success;
     // Set exit result.
     returnValue = success ? EXIT_SUCCESS : EXIT_FAILURE;
-    if(success)
-    {
+    if (success && !ShutdownRequested()) {
```

Comment:
I'm not convinced that this makes a difference at all. initialzeResult is emitted with the return value of `appInitMain`. The return value of `appInitMain` ("success") is supposed to be false when shutdown was requested during initialization.

Reply:
It can make a difference if shutdown is requested between `AppInitMain()` returns and when this slot is executed, it's a QueuedConnection.

Reply:
Maybe 
```diff
diff --git a/src/init.cpp b/src/init.cpp
index 50643ff96b..1f46e5e3d2 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -1736,5 +1736,5 @@ bool AppInitMain()
     StartWallets(scheduler);
 #endif
 
-    return true;
+    return !ShutdownRequested();
 }

Reply:
Indeed `AppInitMain()` return was changed recently in #11831, but I still think that is correct. IMO this change is also correct, it avoids initialization **but** the application should not crash without this change, so I don't agree it's a fix.

Reply:
> It can make a difference if shutdown is requested between AppInitMain() returns and when this slot is executed, it's a QueuedConnection.

Then after AppInitMain was successfully executed, a shutdown was requested (but not yet acted on), so I don't understand why going into qt initialization would crash. It could successfully shutdown after that.

But anyhow if this fixes a reproducible issue I'm okay with it, it's possible that there are interactions I don't follow.

Reply:
> so I don't understand why going into qt initialization would crash

That's why I think this does't fix the problem, just avoids it.

Reply:
That's good enough for rc3, though.